engine.py

```diff
@@ -912,3 +912,58 @@ def __setup_favorite_dirs(self):
                                 type=(nuke.IMAGE | nuke.SCRIPT | nuke.GEO),
                                 icon=icon_path,
                                 tooltip=path)
+
+    #####################################################################################
+    # Script and session related methods
+
+    def get_session_path(self):
+        """
+        Returns the path to the current file if it resides on
+        disk. If unsaved, it returns None.
+
+        :return: Path to the current scene if file is saved, else returns
+                 and empty string.
+        """
+        # retrieve the name of the nuke file. In case of a saved file, the
```

Comment:
Is that compatible with Nuke Studio and Hiero? I don't know how these environments work. @thebeeland would know more.

Reply:
Looks like it is compatible with Nuke Studio but always returns an empty string in Hiero. The closest I could find is the last loaded project path with `hiero.core.projects()[-1].path()` but there should be something better.

Reply:
Please add hiero support before we merge this in!